### PR TITLE
Add shiboken build function  to findpyside script.

### PR DESCRIPTION
**Pull request by sphrose**
Signed-off-by: sphrose <82213493+sphrose@users.noreply.github.com>

Changed pyside and shiboken projects to be shared so that they pass on dependencies correctly, added shiboken_tool target so that the exe can be called, added function to simplify shiboken builds from code.

### CODE REVIEW

File: package-system/pyside2/Findpyside2.cmake.Windows [hunk 1]
```diff
@@ -12,13 +12,40 @@ if (TARGET ${TARGET_WITH_NAMESPACE})
...
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtTest
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtUiTools
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtWidgets
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtWinExtras
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtXml
+)
+
+set_target_properties(${TARGET_WITH_NAMESPACE} PROPERTIES 
+    ${MY_NAME}_SHARE_DIR ${CMAKE_CURRENT_LIST_DIR}/pyside2/share
+    IMPORTED_LOCATION ${SHIBOKEN_SHARED_LIBRARIES} "${${MY_NAME}_LIB_DIR}/libshiboken2.abi3.so"
```
Comment (jackalbe) [inline package-system/pyside2/Findpyside2.cmake.Windows:47]: the cmake file package-system/pyside2/Findpyside2.cmake.Windows indicates Windows, but the set target properties points to an SO file. Is this correct?

Comment (sphrose) [reply package-system/pyside2/Findpyside2.cmake.Windows:47]: Arg, no. Mix up somewhere, thanks.

Comment (sphrose) [reply package-system/pyside2/Findpyside2.cmake.Windows]: Fixed now.

File: package-system/pyside2/Findpyside2.cmake.Linux [hunk 2]
```diff
@@ -25,11 +25,38 @@ set(${MY_NAME}_RUNTIME_DEPENDENCIES
...
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtSql
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtSvg
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtWidgets
+        ${${MY_NAME}_INCLUDE_DIR}/PySide2/QtXml
+)
+
+set_target_properties(${TARGET_WITH_NAMESPACE} PROPERTIES 
+    ${MY_NAME}_SHARE_DIR ${CMAKE_CURRENT_LIST_DIR}/pyside2/share
+    IMPORTED_LOCATION "${${MY_NAME}_LIB_DIR}/libpyside2.abi3.so"
+    IMPORTED_LOCATION_DEBUG "${${MY_NAME}_LIB_DIR}/libpyside2.abi3.so"
```
Comment (spham-amzn) [inline package-system/pyside2/Findpyside2.cmake.Linux:58]: nit:
Might as well use ${CMAKE_SHARED_LIBRARY_SUFFIX} here as well

Comment (sphrose) [reply package-system/pyside2/Findpyside2.cmake.Linux]: Done

**Review by spham-amzn — APPROVED** [review]

File: package-system/pyside2/Findpyside2.cmake.Linux [hunk 3]
```diff
@@ -25,11 +25,38 @@ set(${MY_NAME}_RUNTIME_DEPENDENCIES
     ${PYSIDE_BASE_PATH}/lib/libshiboken2.abi3.so.5.15
     ${PYSIDE_BASE_PATH}/lib/libshiboken2.abi3.so
 )
+
 ly_add_target_files(TARGETS ${TARGET_WITH_NAMESPACE} FILES ${${MY_NAME}_RUNTIME_DEPENDENCIES})
 
 # In addition to the libraries, create a Pyside2::Tools library that handles the shiboken and pyside2-lupdate executables that are part o
 # of the pyside2 package
```
Comment (lawsonamzn) [inline package-system/pyside2/Findpyside2.cmake.Linux:32]: opportunity to fix the typo here (the floating o)

**Review by lawsonamzn — COMMENTED** [review]
The two find files are largely identical except for a few things like dll names and so on.  perhaps they could be combined?

but also, how is this being tested?   You'd have to manually overwrite your find script in your 3rd party packages with this one to use it, or you'd have to deploy a new package.  What's the plan?

File: package-system/pyside2/Findpyside2.cmake.Linux (hunk 3 above)
Comment (sphrose) [reply package-system/pyside2/Findpyside2.cmake.Linux:32]: Yes, combining them works. For testing I've been overwriting the Findpyside.cmake file in my existing 3p pyside package.

Comment (sphrose) [reply package-system/pyside2/Findpyside2.cmake.Linux]: Merged.

**Review by lawsonamzn — APPROVED** [review]
LGTM.   Once this is in we can start rev-ing the package.  will have to plan for that..

I take it no mac support was, or is, provided?

**Comment by sphrose** [comment]
> LGTM. Once this is in we can start rev-ing the package. will have to plan for that..
> 
> I take it no mac support was, or is, provided?

No, there wasn't any in the original pyside and  I don't have a mac to test on.

**Comment by jackalbe** [comment]
Will this be built for Mac at some point?

**Comment by spham-amzn** [comment]
> Will this be built for Mac at some point?

I will see if we can add that to our backlog

**Pull request merged**